pyquil/api/_wavefunction_simulator.py

```diff
@@ -126,11 +139,27 @@ def run_and_measure(self, quil_program: Program, qubits: List[int] = None,
             respected in the returned bitstrings. If not provided, all qubits used in
             the program will be measured and returned in their sorted order.
         :param int trials: Number of times to sample from the prepared wavefunction.
+        :param memory_map: An assignment of classical registers to values, representing an initial
+                           state for the QAM's classical memory.
         :return: An array of measurement results (0 or 1) of shape (trials, len(qubits))
         """
         if qubits is None:
             qubits = sorted(quil_program.get_qubits(indices=True))
 
+        if memory_map is not None:
+            quil_program = self.augment_program_with_memory_values(quil_program, memory_map)
+
         return self.connection._run_and_measure(quil_program=quil_program, qubits=qubits,
                                                 trials=trials,
                                                 random_seed=self.random_seed)
+
+    @staticmethod
+    def augment_program_with_memory_values(quil_program, memory_map):
```

Comment:
don't we already have a function that does this? Does this work with defined gates?

Reply:
answering my own questions: yes, but it is also a staticmethod on a class; and yes (resp)